src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -630,6 +630,10 @@ private static async Task<bool> IsHubMethodAuthorizedSlow(IServiceProvider provi
                         cts = CancellationTokenSource.CreateLinkedTokenSource(connection.ConnectionAborted);
                         arguments[parameterPointer] = cts.Token;
                     }
+                    else if (descriptor.IsServiceArgument(parameterPointer))
```

Comment:
This should be last.

Reply:
Doesn't matter here. The discovery logic though should be moved to last.

Reply:
So if I add a `ChannelReader<T>` to the container it won't override the existing impl right?

Reply:
It will right now, but that's what I was saying about the discovery logic, need to fix that.

Reply:
I see, well lets fix that then.